### PR TITLE
Feature/add examples

### LINKED ISSUE #3: Create documentation
Add documentation for core module functionality

### LINKED ISSUE #4: Add tests
Add tests for core module functionality

### LINKED ISSUE #11: Add example file/dir upload

### PR DESCRIPTION
**Pull request by zhenik**
- [x] doc
- [x] example
- [x] test

### CODE REVIEW

File: dev/ansible/03_run_tests.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    msg: "{{ test_output }}"
+
+- name: Register proxy PID_ID
+  shell: lsof -t -i:9999
+  register: PROXY_PID_ID # kill later with smth like `sudo kill -TERM 4635`
+
+- name: Minio curl example. Verify that proxy to minio works
+  uri:
+    url: http://127.0.0.1:9999
+    status_code: 403
```
Comment (fredrikhgrelland) [inline dev/ansible/03_run_tests.yml:35]: Why 403?

File: dev/ansible/03_run_tests.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  until: curl_output.status == 403
+  retries: 10
+  delay: 5
+
+- name: Debug curl_output
+  debug:
+    msg: "{{curl_output}}"
+
+- name: Add minio config
+  shell: mc config host add I-AM-THE-CONFIG http://127.0.0.1:9999 minio minio123
```
Comment (fredrikhgrelland) [inline dev/ansible/03_run_tests.yml]: I would suggest something like minio-container or similar. I-AM-THE-CONFIG seems out of place.

File: dev/ansible/03_run_tests.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+# kill -TERM 6504
+
+#- name: Add minio config
+#  shell:
+#    cmd: mc config host add LOCALMINIO http://127.0.0.1:9000
+#  register: secret_id
+#- name: Minio bucket `one` exist
+
+
```
Comment (fredrikhgrelland) [inline dev/ansible/03_run_tests.yml]: Remove.

Comment (fredrikhgrelland) [inline dev/ansible/03_run_tests.yml]:
```suggestion
```

File: dev/ansible/proxy-minio.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+consul connect proxy -service minio-local -upstream minio:9999 -log-level debug </dev/null &>/dev/null & PROXY_PID_ID=$!
+echo $PROXY_PID_ID
```
Comment (fredrikhgrelland) [inline dev/ansible/proxy-minio.sh]: Isn't this inline in ansible?

File: variables.tf [hunk 5]
```diff
@@ -11,7 +11,7 @@ variable "nomad_datacenters" {
 }
 variable "nomad_namespace" {
   type        = string
-  description = "[Enterprise] Nomad namespace"
+  description = "[Enterprise feature] Nomad namespace"
```
Comment (fredrikhgrelland) [inline variables.tf]: Nah.. keep it as is.

**Review by fredrikhgrelland — COMMENTED** [review]
Some comments, but otherwise really good. Good spread of examples. We can use this for a lot of smoke tests.

File: dev/ansible/03_run_tests.yml (hunk 1 above)
Comment (zhenik) [reply dev/ansible/03_run_tests.yml:35]: REST api works that way https://github.com/minio/minio/issues/4851#issuecomment-324543717 

Here I wanted to quickly verify that proxy available (without any credentials)

File: dev/ansible/proxy-minio.sh (hunk 4 above)
Comment (zhenik) [reply dev/ansible/proxy-minio.sh]: I left it, due that PR is still a draft. I will remove it, once PR is ready

File: dev/ansible/03_run_tests.yml (hunk 1 above)
Comment (fredrikhgrelland) [reply dev/ansible/03_run_tests.yml:35]: Please add a comment inline to state that fact.

**Review by fredrikhgrelland — CHANGES_REQUESTED** [review]
One tiny comment and ready for rebase.

File: README.md [hunk 6]
```diff
@@ -17,6 +21,11 @@
...
 ```
 
+Command will run an example with standalone instance of minio.
+Minio example instance has:
+- [buckets ["one", "two"]](./example/main.tf)
+- [different type of files uploaded to bucket `one/`](./dev/ansible/04_upload_files.yml)
+
 ### Requirements
 
 #### Required modules
```
Comment (claesgill) [inline README.md:31]: Maybe remove empty sections?

File: README.md [hunk 7]
```diff
@@ -51,20 +60,28 @@ make up
...
 }
 ```
 
 ### Verifying setup
-Description of expected end result and how to check it. E.g. "After a successful run Presto should be available at localhost:8080".
+
+You can verify successful run with next steps:
+
+* create local proxy to minio instance. Require [`consul`](https://releases.hashicorp.com/consul/) binary available on `PATH` of local machine.
+```text
```
Comment (claesgill) [inline README.md]: I think maybe this should be moved to the *Reqired software* section

**Review by claesgill — COMMENTED** [review]
Have some minor comments, but it runs and looks really good. :rocket:

File: README.md (hunk 6 above)
Comment (zhenik) [reply README.md:31]: I would like to leave it as a reminder.

**Pull request merged**